protocols/decoder.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protocols
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/protocols/bytes"
+	"go.opentelemetry.io/collector/protocols/models"
+)
+
+type MetricsDecoder struct {
+	mod models.MetricsDecoder
+	enc bytes.MetricsDecoder
+}
+
+type TracesDecoder struct {
+	mod models.TracesDecoder
+	enc bytes.TracesDecoder
+}
```

Comment:
Move this close to the func, and all of them.

Reply:
👍 though dropping these files from this PR since it needs unit tests which are still WIP.